docs/separating-applications.md

```diff
@@ -0,0 +1,66 @@
+Typically, you would have a single package.json with a single webpack config that produces a bundle
+that can be included in an html file with a `<script>` tag.
+
+Advantages:
+- Simplest to set up
+- [monolithic version control has some advantages](https://danluu.com/monorepo/)
+
+Disadvantages:
+- One master webpack config and package.json means less flexibility / freedom for individual projects
+- Slow build times once your project gets large
+- Builds and deployments are all tied together, which can necessitate fixed release schedules instead of ad hoc releases.
+
+#### Option 2: NPM packages
+
+Create a root application that npm installs each of the single-spa applications. Each child application
+is in a separate code repository and is responsible for publishing a new version everytime that it updates.
+The root application should reinstall, rebuild, and redeploy whenever a single-spa application changes.
+
+Typically, the single-spa applications compile themselves separately with babel and/or webpack.
+
+Note that you can also use the [monorepo methodology](https://medium.com/netscape/the-case-for-monorepos-907c1361708a) which
+allows for separate builds without having separate code repositories.
+
+Advantages:
+- npm install is familiar and easy to set up
+- Separate npm packages means each application can build itself separately before publishing to npm
+
+Disadvantages:
+- The root application must reinstall the child applications in order to rebuild/redeploy
+- Medium difficulty to set up
+
+#### Option 3: Dynamic Module Loading
+
+Create a root application which can allow single-spa applications to deploy themselves separately. To do so,
+create a manifest file that the single-spa applications update during their deployment process, which controls
+which versions of the single-spa applications are "live". Then change which javascript file is loaded based on the manifest.
+
+Changing which javascript file is loaded for each child application can be done in many ways.
+1) Web server: have your webserver create a dynamic script tag for the "live" version of each single-spa application.
+2) Use a [module loader](https://www.jvandemo.com/a-10-minute-primer-to-javascript-modules-module-formats-module-loaders-and-module-bundlers/)
+   such as [SystemJS](https://github.com/systemjs/systemjs) that can download and execute javascript code in the browser
+   from dynamic urls.
+
+#### Comparison
+|                | Separate code repositories | Separate builds | Separate deployments | Difficulty to set up
+| -------------- | -------------------------- | --------------- | -------------------- | --------------------
+| Monorepo       |                            |                 |                      |           1
+| NPM modules    |            [x]             |       [x]       |                      |           2
+| Module loading |            [x]             |       [x]       |          [x]         |           3
```

Comment:
I like the idea of difficulty to setup column, but it's unclear at a glance if 1=hardest or 1=easiest. Maybe it would be easier to just put 'easy', 'medium', 'hard' or something (which also matches what you have written in the options sections).

Also, another potentially pointless suggestion, but maybe move the comparison to the top? Everyone likes a good TLDR before they dive into a topic.

Reply:
Good idea, I'll change to easy/medium/hard. The comparison already is in the sections above, but I restate it here.